micrometer-core/src/main/java/io/micrometer/core/instrument/observation/DefaultMeterObservationHandler.java

```diff
@@ -65,8 +65,8 @@ public void onStop(Observation.Context context) {
     }
 
     @Override
-    public boolean supportsContext(Observation.Context context) {
-        return true;
+    public void onEvent(Observation.Event event, Observation.Context context) {
+        Counter.builder(event.getName()).tags(createTags(context)).register(meterRegistry).increment();
```

Comment:
Since the event is scoped to the Observation, I wonder if the Counter name should somehow include both. observation name + event name, perhaps? I'm not sure how this looks given typical use cases, though, which is probably the best gauge of what makes sense. Unless events are supposed to be standalone things that just happen to occur during an observation?

Reply:
It makes a lot of sense to me. Since we are attaching an Event to an Observation, I think it is a fair assumption that they belong together.

I can imagine a use-case where Events are just happening standalone, not connected to an Observation but to cover that use-case, I would use a different component (if we want to cover that at all).

A real-life example I can think of is applying the same business logic for two different endpoints or sources: let's say we are accepting messages over HTTP and AMQP but the workflow/business logic is the same for both.